File: finder4j-backtrack-examples/src/test/java/com/github/k0zka/finder4j/backtrack/examples/queens/QueensStepFactoryTest.java
package com.github.k0zka.finder4j.backtrack.examples.queens;

import java.util.List;

import org.junit.Assert;
import org.junit.Test;

import com.github.k0zka.finder4j.backtrack.examples.queens.QueensState;
import com.github.k0zka.finder4j.backtrack.examples.queens.QueensStep;
import com.github.k0zka.finder4j.backtrack.examples.queens.QueensStepFactory;

public class QueensStepFactoryTest {
	@Test
	public void produce() {
		List<QueensStep> steps = new QueensStepFactory().produce(new QueensState((short)1, (short)1, null, null));
		Assert.assertFalse(steps.isEmpty());
	}
}

File: finder4j-backtrack-examples/src/test/kotlin/com/github/k0zka/finder4j/backtrack/examples/queens/QueensStepFactoryTest.kt
package com.github.k0zka.finder4j.backtrack.examples.queens

import com.github.k0zka.finder4j.backtrack.examples.queens.QueensState.Companion.notSet
import org.junit.Assert
import org.junit.Test

class QueensStepFactoryTest {
	@Test
	fun produce() {
		val steps = QueensStepFactory().produce(
				QueensState(arrayOf(1.toShort(), 1.toShort(), notSet, notSet).toShortArray()))
		Assert.assertFalse(steps.isEmpty())
	}
}
